pkg/controller/handlers/toolreference/migrate.go

```diff
@@ -21,6 +21,11 @@ func (h *Handler) MigrateToolNames(req router.Request, _ router.Response) error
 		tools = o.Spec.Manifest.Tools
 	case *v1.Workflow:
 		tools = o.Spec.Manifest.Tools
+	case *v1.Thread:
```

Comment:
Processing threads will likely have an impact on controller, if so we can store the hash of the migration map into each object and compare to only run migration once map is changed. Although I feel like the speed impact shouldn't be that bad.